File: go/tree/508. 出现次数最多的子树元素和.go
package tree

func findFrequentTreeSum(root *TreeNode) []int {
	var dfs func(*TreeNode) int
	d := make(map[int]int)
	var res []int
	var mostFreq int

	dfs = func(root *TreeNode) int {
		if root == nil {
			return 0
		}
		sum := root.Val + dfs(root.Left) + dfs(root.Right)
		d[sum]++
		if d[sum] > mostFreq {
			mostFreq = d[sum]
		}
		return sum
	}

	dfs(root)

	for k, v := range d {
		if v == mostFreq {
			res = append(res, k)
		}
	}
	return res
}

File: python/树/508. 出现次数最多的子树元素和.py
from typing import List

class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def findFrequentTreeSum(self, root: TreeNode) -> List[int]:
        if not root: return []
        from collections import defaultdict
        d = defaultdict(int)
        def dfs(root):
            if not root:
                return 0
            left = dfs(root.left)
            right = dfs(root.right)
            sum = left + right + root.val
            d[sum] +=1
            return sum
        dfs(root)
        tmp = sorted(d.items(),key=lambda x:x[1],reverse=True)
        return [k for k,v in tmp if v == tmp[0][1]]